File: test_gemini.py
import google.generativeai as genai
import os
from dotenv import load_dotenv

# Path to the .env file in the backend directory
env_path = os.path.join(os.path.dirname(__file__), 'backend', '.env')
load_dotenv(dotenv_path=env_path)

# Retrieve the API key and model from environment variables
api_key = os.getenv("GOOGLE_API_KEY")
model_name = os.getenv("GEMINI_MODEL", "gemini-1.5-flash")

if not api_key:
    print("Error: GOOGLE_API_KEY not found in backend/.env")
else:
    # Configure the Gemini API
    genai.configure(api_key=api_key)

    # Initialize the Gemini model
    model = genai.GenerativeModel(model_name)

    # Generate text
    try:
        response = model.generate_content("Write a short poem about artificial intelligence.")
        print(response.text)
    except Exception as e:
        print(f"Error occurred: {e}")

File: backend/test-gemini.js
import { GoogleGenerativeAI } from "@google/generative-ai";
import dotenv from 'dotenv';
dotenv.config();

const genAI = new GoogleGenerativeAI(process.env.GOOGLE_API_KEY);

async function testModel(modelName) {
  try {
    console.log(`Testing ${modelName}:`);
    const model = genAI.getGenerativeModel({ model: modelName });
    const result = await model.generateContent("Hi!");
    const response = await result.response;
    console.log(`SUCCESS for ${modelName}:`, response.text());
  } catch (err) {
    console.error(`FAIL for ${modelName}:`, err.message);
  }
}

async function run() {
  const models = ["gemini-1.5-flash", "gemini-1.5-flash-latest", "gemini-pro", "gemini-1.5-pro"];
  for (const m of models) {
    try {
      await testModel(m);
    } catch (e) {}
  }
}

run();


